src/tags/object/Video/Rectangle.js

```diff
@@ -22,45 +31,175 @@ const RectanglePure = ({ reg, frame, workingArea, ...rest }) => {
+          topRight = Math.min(cornerRadius[1], width / 2, height / 2);
+          bottomRight = Math.min(cornerRadius[2], width / 2, height / 2);
+          bottomLeft = Math.min(cornerRadius[3], width / 2, height / 2);
+        }
+        context.moveTo(topLeft, 0);
+        context.lineTo(width - topRight, 0);
+        context.arc(width - topRight, topRight, topRight, (Math.PI * 3) / 2, 0, false);
+        context.lineTo(width, height - bottomRight);
+        context.arc(width - bottomRight, height - bottomRight, bottomRight, 0, Math.PI / 2, false);
+        context.lineTo(bottomLeft, height);
+        context.arc(bottomLeft, height - bottomLeft, bottomLeft, Math.PI / 2, Math.PI, false);
+        context.lineTo(0, topLeft);
+        context.arc(topLeft, topLeft, topLeft, Math.PI, (Math.PI * 3) / 2, false);
+      }
+      context.closePath();
+      context.fillStrokeShape(shape);
+    },
+    [isSticking, maxWidth],
+  );
+
+  const groupBox = {
+    strokeScaleEnabled: false,
+  };
 
   const newBox = {
-    x: box.x * waWidth / 100,
-    y: box.y * waHeight / 100,
-    width: box.width * waWidth / 100,
-    height: box.height * waHeight / 100,
+    x: (box.x * waWidth) / 100,
+    y: (box.y * waHeight) / 100,
+    width: (box.width * waWidth) / 100,
+    height: (box.height * waHeight) / 100,
     rotation: box.rotation,
   };
 
+  const onClickLabel = reg.onCLickRegion;
+
   return reg.isInLifespan(frame) ? (
-    <Rect
-      {...newBox}
-      draggable
-      fill={style.fillColor}
-      stroke={style.strokeColor}
-      strokeScaleEnabled={false}
-      opacity={reg.hidden ? 0 : 1}
-      onTransformEnd={e => {
-        const node = e.target;
-        const scaleX = node.scaleX();
-        const scaleY = node.scaleY();
-        // set minimal value
-        const w = Math.max(3, node.width() * scaleX);
-        const h = Math.max(3, node.height() * scaleY);
-
-        node.scaleX(1);
-        node.scaleY(1);
-        node.setWidth(w);
-        node.setHeight(h);
-
-        reg.updateShape(getNodeAbsoluteDimensions(node, workingArea), frame);
-      }}
-      onDragEnd={e => {
-        const node = e.target;
-
-        reg.updateShape(getNodeAbsoluteDimensions(node, workingArea), frame);
-      }}
-      {...rest}
-    />
+    <Group {...groupBox}>
+      {ss && (
+        <Label
+          x={newBox.x}
+          y={newBox.y - height}
+          scaleX={scale}
+          scaleY={scale}
+          onClick={() => {
+            return false;
+          }}
+        >
+          <Tag fill={Utils.Colors.getScaleGradient(score)} cornerRadius={2} />
+          <Text
+            text={score.toFixed(2)}
+            fontFamily="-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, 'Helvetica Neue', Arial, 'Noto Sans', sans-serif"
+            fontSize={fontSize}
+            fill="white"
+            padding={2}
+          />
+        </Label>
+      )}
+      {showLabels && (
```

Comment:
Let's do this better - move this code to separate component and just use it here